packages/react/src/themes/teams/components/Text/textVariables.ts

```diff
@@ -1,4 +1,15 @@
 import { TextVariables } from '../../../base/components/Text/textVariables'
+import { StrictColorSchemeMapping, StrictColorScheme } from '../../../../themes/types'
+import { TeamsColorNames } from '../../colors'
+import { generateComponentAreas } from '../../../colorUtils'
+
+const textColorComponentAreas = generateComponentAreas('foreground')
+export type TextColorComponentAreas = typeof textColorComponentAreas[number]
+
+export type TextColorSchemeMapping = StrictColorSchemeMapping<
```

Comment:
agreed to simplify the set of typings that are provided as part of client's API, to ensure that client needs to specify only those type aspects that actually change (names of types are subject to change):

```jsx
// this is the simplest type that could be used on the client side
type TeamsSchemeMapping<TAreas extends string> =  StrictColorSchemeMapping<
  StrictColorScheme<TAreas>,
  TeamsColorNames
>

type ItemType<T> = T extends (infer TItem)[] ? TItem : never
```

Then, to simply define scheme mapping for Teams, on the client side it will be sufficient to use the following type declaration:

```jsx
type TextColorSchemeMapping = TeamsSchemeMapping<'foreground'> 

// or, with more explicit naming strategy for the same type
type TextColorSchemeMapping = TeamsSchemesWithAreas<'foreground'>
```

and, in case if item type of runtime array is intended to be used, then

```jsx
const textColorAreas = stringLiteralsArray('foreground')
type TextColorSchemeMapping = TeamsSchemesWithAreas<ItemType<typeof textColorAreas>>
```